server/ott-config.ts

```diff
@@ -533,24 +558,24 @@ function postProcessConfig(): void {
 	}
 }
 
-function validateMail() {
+function validateMail(): Result<void, Error> {
 	const sender = conf.get("mail.sender_email");
+
 	if (sender && !validator.isEmail(sender)) {
-		log.error(`Invalid email address ${sender} found in config, disabling mail.`);
-		conf.set("mail.enabled", false);
-		return;
+		return err(new Error(`Invalid email address ${sender} found in config, disabling mail.`));
 	}
 
 	if (!conf.get("mail.mailjet_api_key") || !conf.get("mail.mailjet_api_secret")) {
-		log.error("Mailjet API key and secret are required to send mail, disabling mail.");
-		conf.set("mail.enabled", false);
-		return;
+		return err(
+			new Error("Mailjet API key and secret are required to send mail, disabling mail.")
+		);
 	}
 
 	if (conf.get("env") === "test") {
-		log.warn("Mail is enabled in test mode. Forcing sandbox mode.");
-		conf.set("mail.mailjet_sandbox", true);
+		return err(new Error("Mail is enabled in test mode. Forcing sandbox mode."));
 	}
```

Comment:
This is not equivalent functionality. Move this check somewhere else and restore the original behavior.